### PR TITLE
Standalone Mode

### PR DESCRIPTION
**Pull request by hoco**
Add standalone-mode for non-ElastiCache memcached environment.

### CODE REVIEW

File: lib/memcached/elasticache.rb [hunk 1]
```diff
@@ -14,22 +14,25 @@ class Client
       # @param [String] config_endpoint ElastiCache config endpoint strings like "my-host.cache.aws.com:11211"
       # @option [Integer] :refresh_interval second-scale interval of refreshing cluster endpoints
       # @option [Integer] :max_retry_count max retry times to command and refresh cluster endpoints
+      # @option [Boolean] :local_mode if local_mode is true, the client always treats the config endpoint as the only cluster endpoint (for non-ElastiCache memcached)
```
Comment (DianthuDia) [inline lib/memcached/elasticache.rb]: [nis]s/local_mode/standalone_mode

File: lib/memcached/elasticache/auto_discovery/endpoint.rb [hunk 2]
```diff
@@ -47,8 +48,13 @@ def get_stats_from_remote
         end
 
         def get_config_from_remote
-          data = remote_command(CONFIG_COMMAND)
-          ConfigResponse.new(data)
+          if @options[:local_mode]
+            # IP address is not used
+            ConfigResponse.new("#{@host}|0.0.0.0|#{@port}")
```
Comment (DianthuDia) [inline lib/memcached/elasticache/auto_discovery/endpoint.rb]: [Q]Why don't you resolve `@host` to Ip address? It more reality.

File: lib/memcached/elasticache.rb (hunk 1 above)
Comment (DianthuDia) [reply lib/memcached/elasticache.rb]: [nits]if standalone_mode is true, the client treats `config_endpoint` as standard memcached server (NO require ElastiCache endpint). It is useful during development.

**Review by DianthuDia — CHANGES_REQUESTED** [review]
👀

**Comment by hoco** [comment]
👍

File: lib/memcached/elasticache/auto_discovery/endpoint.rb [hunk 3]
```diff
@@ -47,8 +48,13 @@ def get_stats_from_remote
         end
 
         def get_config_from_remote
-          data = remote_command(CONFIG_COMMAND)
-          ConfigResponse.new(data)
+          if @options[:standalone_mode]
+            # IP address is not used
```
Comment (DianthuDia) [inline lib/memcached/elasticache/auto_discovery/endpoint.rb]: [IMO] 🔥 🍚

**Pull request merged**